doc/sphinx/source/recipes/recipe_smpi.rst

```diff
@@ -0,0 +1,93 @@
+Single Model Perfomance Index (SMPI)
+====================================
+
+Overview
+--------
+
+This diagnostic calculates the Single Model Performance Index (SMPI) following Reichler and Kim (2008). The SMPI (called "I\ :sup:`2`") is based on the comparison of several different climate variables (atmospheric, surface and oceanic) between climate model simulations and observations or reanalyses, and it focuses on the validation of the time-mean state of climate. For I\ :sup:`2` to be determined, the differences between the climatological mean of each model variable and observations at each of the available data grid points are calculated, and scaled to the interannual variance from the validating observations. This interannual variability is determined by performing a bootstrapping method (random selection with replacement) for the creation of a large synthetic ensemble of observational climatologies. The results are then scaled to the average error from a reference ensemble of models, and in a final step the mean over all climate variables and one model is calculated. The plot shows the I\ :sup:`2` values for each model (orange circles) and the multi-model mean (black circle), with the diameter of each circle representing the range of I\ :sup:`2` values encompassed by the 5th and 95th percentiles of the bootstrap ensemble. The I\ :sup:`2` values vary around one, with values greater than one for underperforming models, and values less than one for more accurate models. 
+
+Note: The SMPI diagnostic needs all indicated variables from all added models for exactly the same time period to be calculated correctly. If one model does not provide a specific variable, either that model cannot be added to the SMPI calculations, or the missing variable has to be removed from the diagnostics all together. 
+
+Available recipes and diagnostics
+-----------------------------------
+
+Recipes are stored in recipe/
+
+* recipe_reichlerkim08bams.yml
+
+Diagnostics are stored in diag_scripts/perfmetrics/
+
+* main.ncl: calculates and (optionally) plots annual/seasonal cycles, zonal means, lat-lon fields and time-lat-lon fields. The calculated fields can also be plotted as difference w.r.t. a given reference dataset. main.ncl also calculates RMSD, bias and taylor metrics. Input data have to be regridded to a common grid in the preprocessor. Each plot type is created by a separated routine, as detailed below.
+* cycle_zonal.ncl: calculates single model perfomance index (Reichler and Kim, 2008). It requires fields precalculated by main.ncl.
+* collect.ncl: collects the metrics previously calculated by cycle_latlon.ncl and passes them to the plotting functions.
+
+User settings
+-------------
+
+#. main.ncl
+
+   *Required settings for script*
+
+   * plot_type: cycle (time), zonal (plev, lat), latlon (lat, lon), cycle_latlon (time, lat, lon)
+   * time_avg: type of time average (opt argument of time_operations in diag_scripts/shared/statistics.ncl)
```

Comment:
Either list the available options here, or solve #809 and add a link to the documentation of that function here, so users can click it. Expecting users to go and read your source code is nowhere near user friendly.

Reply:
I have adopted this description directly from perfmetrics, so that the descriptions are consistent.

Reply:
The documentation of perfmetrics should also be fixed, that's why I opened issue #809 in the first place.

Reply:
Agree, but better in another PR.

Reply:
That's fine, but the documentation for the magic project should be ready soon. Who is going to fix that issue and when?

Reply:
The issue was originally assigned to @axel-lauer because he took care of this in version 1.